src/Umbraco.Infrastructure/Runtime/FileSystemMainDomLock.cs

```diff
@@ -132,7 +132,7 @@ private void ListeningLoop()
                 break;
             }
 
-            Thread.Sleep(_globalSettings.CurrentValue.MainDomReleaseSignalPollingInterval);
```

Comment:
I know this is merged already, but this should really pass in the CancellationToken.

Reply:
@PhyxionNL Thanks for the feedback 👍

Feel free to create your own PR and we'll gladly review it 😁

Reply:
Oh boy, we've chosen the red pill here...

Let me ask you: "What cancellation token"!?!?!

The one that comes from the  `private readonly CancellationTokenSource`...? 
The one that's not referenced from anything else... so is never _actually_ cancelled!?!